src/generate.zig

```diff
@@ -1,30 +1,35 @@
 const std = @import("std");
 
-
 // ascii mapping
 const A = 65;
 const z = 122;
 
-
-
-pub fn generateField(random: std.rand.Random, comptime T: type) T {  
+pub fn generateField(alloc: std.mem.Allocator, random: std.rand.Random, comptime T: type) std.mem.Allocator.Error!T {
     const typeInfo = @typeInfo(T);
     switch (typeInfo) {
+        .Bool => {
+            return random.boolean();
+        },
         .Int => {
-            const len: comptime_int = if (T == u8) 4 else if (T == u32) 4 else 7;
-            var boundaries: [len]T =  undefined;
+            const len: comptime_int = if (typeInfo.Int.signedness == .signed) 7 else 4;
+            var boundaries: [len]T = undefined;
             getBoundaries(T, &boundaries);
             if (random.float(f32) < 0.2) {
                 return boundaries[random.intRangeAtMost(usize, 0, boundaries.len - 1)];
             } else {
                 if (T == u8) {
                     return random.intRangeAtMost(T, A, z);
+                } else if (T == i8) {
+                    return random.intRangeAtMost(T, std.math.minInt(T), std.math.maxInt(T));
                 } else if (T == u32) {
-                    const max = std.math.pow(T, 10, 3);
+                    const max = comptime std.math.pow(T, 10, 3);
+                    return random.intRangeAtMost(T, 0, max);
+                } else if (typeInfo.Int.signedness == .unsigned) {
+                    const max = comptime std.math.maxInt(T);
                     return random.intRangeAtMost(T, 0, max);
                 } else {
-                    const min = -std.math.pow(T, 10, 3);
-                    const max = std.math.pow(T, 10, 3);
+                    const min = comptime -std.math.pow(T, 10, 3);
```

Comment:
Oh, I made some of these `comptime` as it seemed to turn a runtime error into a compile time error.  Might not be right about that, but it's effectively a static value in any case and i was hitting errors on negative numbers for unsigned values (which is what led to the code above).